### PR TITLE
Spanner to Spanner additional cases as per ITN.

### PR DESCRIPTION
**Pull request by NavdeepCS**
Spanner to Spanner Additional cases as per ITN E2E coverage.

### CODE REVIEW

File: src/e2e-test/features/spanner/source/SpannerToSpanner_Additional.feature [hunk 1]
```diff
@@ -0,0 +1,118 @@
+@Spanner_Source @SPANNER_TEST
```
Comment (bharatgulati) [inline src/e2e-test/features/spanner/source/SpannerToSpanner_Additional.feature:1]: What does @SPANNER_TEST do to the feature, and why isn't this tag present in the stepdesign package?

File: src/e2e-test/features/spanner/source/SpannerToSpanner_Additional.feature [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+Feature: Spanner source - Verification of Additional Spanner to Spanner successful data transfer without using connections
```
Comment (bharatgulati) [inline src/e2e-test/features/spanner/source/SpannerToSpanner_Additional.feature:2]: remove **without using connection** text from the feature name as we already providing this details in the scenario name

File: src/e2e-test/features/spanner/source/SpannerToSpanner_Additional.feature (hunk 1 above)
Comment (NavdeepCS) [reply src/e2e-test/features/spanner/source/SpannerToSpanner_Additional.feature:1]: It creates a random Spanner instance,database and table with 7 entries at the source.
It is an existing hook whose code was not altered for this PR.

File: src/e2e-test/features/spanner/source/SpannerToSpanner_Additional.feature [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+
+  @SPANNER_TEST @EXISTING_SPANNER_SINK
+  Scenario: To verify data is getting transferred from Spanner source to existing Spanner sink without using connection functionality
```
Comment (bharatgulati) [inline src/e2e-test/features/spanner/source/SpannerToSpanner_Additional.feature:5]: Change the scenario name and provide more clear statement. 
_To verify data is getting transferred from Spanner source to Spanner sink having existing table name without using connection functionality_

File: src/e2e-test/features/spanner/source/SpannerToSpanner_Additional.feature [hunk 4]
```diff
@@ -0,0 +1,118 @@
...
+    Then Wait till pipeline preview is in running state
+    Then Open and capture pipeline preview logs
+    Then Verify the preview run status of pipeline in the logs is "succeeded"
+    Then Close the pipeline logs
+    Then Close the preview
+    Then Deploy the pipeline
+    Then Run the Pipeline in Runtime
+    Then Wait till pipeline is in running state
+    Then Verify the pipeline status is "Succeeded"
+    Then Validate records transferred to existing target spanner table with record counts of  source spanner table
```
Comment (bharatgulati) [inline src/e2e-test/features/spanner/source/SpannerToSpanner_Additional.feature:42]: Provide more clear statement. For example, 
Validate the data transferred from spanner source to existing table of spanner sink.

File: src/e2e-test/features/spanner/source/SpannerToSpanner_Additional.feature [hunk 5]
```diff
@@ -0,0 +1,118 @@
+@Spanner_Source @SPANNER_TEST
+Feature: Spanner source - Verification of Additional Spanner to Spanner successful data transfer without using connections
+
+  @SPANNER_TEST @EXISTING_SPANNER_SINK
```
Comment (bharatgulati) [inline src/e2e-test/features/spanner/source/SpannerToSpanner_Additional.feature:4]: @SourceRequired Tag is missing, Please identify if this test is also required or not.

File: src/e2e-test/features/spanner/source/SpannerToSpanner_Additional.feature [hunk 6]
```diff
@@ -0,0 +1,118 @@
...
+    Then Close the pipeline logs
+    Then Close the preview
+    Then Deploy the pipeline
+    Then Run the Pipeline in Runtime
+    Then Wait till pipeline is in running state
+    Then Verify the pipeline status is "Succeeded"
+    Then Validate records transferred to existing target spanner table with record counts of  source spanner table
+
+  @SPANNER_SINK_TEST @SPANNER_TEST @Spanner_Source_Required
+  Scenario: To verify data is getting transferred from NoSchema Spanner source to Spanner non existing sink without using connection functionality
```
Comment (bharatgulati) [inline src/e2e-test/features/spanner/source/SpannerToSpanner_Additional.feature:45]: Revise the scenario name

File: src/e2e-test/features/spanner/source/SpannerToSpanner_Additional.feature (hunk 1 above)
Comment (bharatgulati) [reply src/e2e-test/features/spanner/source/SpannerToSpanner_Additional.feature:1]: Remove the SPANNER_TEST Tag from the feature level and only use the tag at scenario level.

File: src/e2e-test/features/spanner/source/SpannerToSpanner_Additional.feature [hunk 7]
```diff
@@ -0,0 +1,118 @@
...
+    Then Wait till pipeline preview is in running state
+    Then Open and capture pipeline preview logs
+    Then Verify the preview run status of pipeline in the logs is "succeeded"
+    Then Close the pipeline logs
+    Then Close the preview
+    Then Deploy the pipeline
+    Then Run the Pipeline in Runtime
+    Then Wait till pipeline is in running state
+    Then Verify the pipeline status is "Succeeded"
+    Then Validate records transferred to target spanner table with record counts of source spanner table
```
Comment (bharatgulati) [inline src/e2e-test/features/spanner/source/SpannerToSpanner_Additional.feature:80]: Are we validating the count here or the data validation?

File: src/e2e-test/features/spanner/source/SpannerToSpanner_withConnections.feature [hunk 8]
```diff
@@ -163,3 +163,52 @@ Feature: Spanner source - Verification of Spanner to Spanner successful data tra
     Then Wait till pipeline is in running state
     Then Verify the pipeline status is "Succeeded"
     Then Validate records transferred to target spanner table with record counts of source spanner table
+
+  @EXISTING_SPANNER_SINK @EXISTING_SPANNER_CONNECTION @Spanner_Source_Required
+  Scenario: To verify data is getting transferred from Spanner source  to existing Spanner sink with use connection functionality
```
Comment (bharatgulati) [inline src/e2e-test/features/spanner/source/SpannerToSpanner_withConnections.feature:168]: Change the scenario outline name

File: src/e2e-test/features/spanner/source/SpannerToSpanner_Additional.feature [hunk 9]
```diff
@@ -0,0 +1,118 @@
...
+
+  @SPANNER_TEST @EXISTING_SPANNER_SINK
+  Scenario: To verify data is getting transferred from NoSchema Spanner source to existing Spanner sink without using connection functionality
```
Comment (bharatgulati) [inline src/e2e-test/features/spanner/source/SpannerToSpanner_Additional.feature:83]: **existing spanner sink** in the outline is not clear, are we using the existing table or database at sink side? If are we utilizing it, then it should be mention in the scenario name.

File: src/e2e-test/features/spanner/source/SpannerToSpanner_Additional.feature (hunk 1 above)
Comment (NavdeepCS) [reply src/e2e-test/features/spanner/source/SpannerToSpanner_Additional.feature:1]: Noted

File: src/e2e-test/features/spanner/source/SpannerToSpanner_Additional.feature (hunk 2 above)
Comment (NavdeepCS) [reply src/e2e-test/features/spanner/source/SpannerToSpanner_Additional.feature:2]: Noted

File: src/e2e-test/features/spanner/source/SpannerToSpanner_Additional.feature (hunk 3 above)
Comment (NavdeepCS) [reply src/e2e-test/features/spanner/source/SpannerToSpanner_Additional.feature:5]: Noted

File: src/e2e-test/features/spanner/source/SpannerToSpanner_Additional.feature (hunk 4 above)
Comment (NavdeepCS) [reply src/e2e-test/features/spanner/source/SpannerToSpanner_Additional.feature:42]: Noted

File: src/e2e-test/features/spanner/source/SpannerToSpanner_Additional.feature (hunk 5 above)
Comment (NavdeepCS) [reply src/e2e-test/features/spanner/source/SpannerToSpanner_Additional.feature:4]: as per discussion with Ankit, @Spanner_Source will include this test . (?)

File: src/e2e-test/features/spanner/source/SpannerToSpanner_Additional.feature (hunk 6 above)
Comment (NavdeepCS) [reply src/e2e-test/features/spanner/source/SpannerToSpanner_Additional.feature:45]: noted

File: src/e2e-test/features/spanner/source/SpannerToSpanner_Additional.feature (hunk 7 above)
Comment (NavdeepCS) [reply src/e2e-test/features/spanner/source/SpannerToSpanner_Additional.feature:80]: both ... records should match in source and sink

File: src/e2e-test/features/spanner/source/SpannerToSpanner_withConnections.feature (hunk 8 above)
Comment (NavdeepCS) [reply src/e2e-test/features/spanner/source/SpannerToSpanner_withConnections.feature:168]: noted

File: src/e2e-test/features/spanner/source/SpannerToSpanner_Additional.feature (hunk 9 above)
Comment (NavdeepCS) [reply src/e2e-test/features/spanner/source/SpannerToSpanner_Additional.feature:83]: we are using database as the same as source , but create a table at the sink using query.

Comment (NavdeepCS) [reply src/e2e-test/features/spanner/source/SpannerToSpanner_Additional.feature:83]: noted

**Pull request closed without merging**